core/storage/file/gae_models.py

```diff
@@ -45,34 +45,96 @@ class FileMetadataModel(base_models.VersionedModel):
 
     @classmethod
     def get_new_id(cls, entity_name):
+        """Raise an exception if an id is not created.
+
+        Args:
+            entity_name: str. An entity name that is tied to the created id.
+        """
         raise NotImplementedError
 
     @classmethod
     def get_undeleted(cls):
+        """Retrieve values that have not been deleted in the file metadata
+        model.
+
+        Returns:
+            A list of undeleted items from the query.
+        """
         return cls.get_all().filter(cls.deleted == False).fetch(  # pylint: disable=singleton-comparison
             feconf.DEFAULT_QUERY_LIMIT)
 
     @classmethod
     def _construct_id(cls, exploration_id, filepath):
+        """ Builds the path for the id to be generated.
+
+        Args:
+            exploration_id: str. The ID of the exploration.
+            filepath: str. The path to the exploration file.
+
+        Returns:
+            The path of the id to be generated.
+        """
         return utils.vfs_construct_path('/', exploration_id, filepath)
 
     @classmethod
     def create(cls, exploration_id, filepath):
+        """ Creates a new file meta data model entry.
+
+        Args:
+            exploration_id: str. The ID of the exploration.
+            filepath: str. The path to the exploration file.
+
+        Returns:
+            str. Instance of the new file meta data model entry.
+        """
         model_id = cls._construct_id(exploration_id, filepath)
         return cls(id=model_id, deleted=False)
 
     @classmethod
     def get_model(cls, exploration_id, filepath, strict=False):
+        """ Gets the model of an exploration by id.
+
+        Args:
+            exploration_id: str. The ID of the exploration.
+            filepath: str. The path to the exploration file.
+            strict: bool. Not raise an error if FileMetadataModel
```

Comment:
This is unclear. See documentation elsewhere for 'strict'; we use this construct elsewhere in the codebase.